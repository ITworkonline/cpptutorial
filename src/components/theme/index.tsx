const baseTheme = {
  fonts: {
    mono: '"SF Mono", "Roboto Mono", Menlo, monospace',
  },
};

const palette = {
  white: "#ffffff",
  light: "#E2E4E9",
  grey1: "#C4C9D4",
  grey2: "#A7AEBE",
  grey3: "#8A93A8",
  grey4: "#6C7893",
  grey5: "#576075",
  grey6: "#414858",
  dark: "#2B303B",
  black: "#0B0C0F",
  blue1: "#0B1A28",
  blue2: "#08365E",
  blue3: "#224F77",
  blue4: "#2D6A9F",
  blue5: "#3884C7",
  blue6: "#609DD2",
  blue7: "#88B5DD",
  blue8: "#AFCEE9",
  blue9: "#D7E6F4",
};

const lightTheme = {
  ...baseTheme,
  colors: {
    background: palette.white,
    headerBg: palette.light,
    icon: palette.blue5,
    iconHover: palette.blue6,
    buttonBg: "#ddd",
    buttonBg2: "#ccc",
    heading: "#000",
    primaryText: "#3B454E",
    preFormattedText: "rgb(245, 247, 249)",
    link: palette.blue5,
    highlight: palette.blue5,
    sidebarHighlight: palette.blue5,
    search: palette.blue5,
    searchPlaceholder: palette.dark,
  },
};

export type Theme = typeof lightTheme;

const darkTheme: Theme = {
  ...baseTheme,
  colors: {
    background: palette.dark,
    headerBg: palette.black,
    icon: palette.blue5,
    iconHover: palette.blue6,
    buttonBg: "#0f4c75",
    buttonBg2: "#34495e",
    heading: "#fff",
    primaryText: "#fff",
    preFormattedText: "#000",
    link: palette.blue5,
    highlight: palette.blue5,
    sidebarHighlight: palette.blue3,
    search: palette.blue5,
    searchPlaceholder: palette.light,
  },
};

export { lightTheme, darkTheme };
