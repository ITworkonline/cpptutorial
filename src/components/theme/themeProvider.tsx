import * as React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import { lightTheme, darkTheme } from "./index";
import Header from "../header";

interface ThemeProviderProps {
  children: React.ReactNode;
}

interface ThemeProviderState {
  isDarkThemeActive: boolean;
}

class ThemeProvider extends React.Component<ThemeProviderProps, ThemeProviderState> {
  state = {
    isDarkThemeActive: false,
  };

  componentDidMount() {
    this.retrieveActiveTheme();
  }

  retrieveActiveTheme = () => {
    const isDarkThemeActiveJson = window.localStorage.getItem("isDarkThemeActive");

    const isDarkThemeActive = isDarkThemeActiveJson ? JSON.parse(isDarkThemeActiveJson) : false;

    this.setState({ isDarkThemeActive });
  };

  toggleActiveTheme = () => {
    this.setState(prevState => ({ isDarkThemeActive: !prevState.isDarkThemeActive }));

    window.localStorage.setItem("isDarkThemeActive", JSON.stringify(!this.state.isDarkThemeActive));
  };

  render() {
    const { children } = this.props;

    const { isDarkThemeActive } = this.state;

    const currentActiveTheme = isDarkThemeActive ? darkTheme : lightTheme;

    return (
      <div>
        <EmotionThemeProvider theme={currentActiveTheme}>
          <Header
            isDarkThemeActive={isDarkThemeActive}
            toggleActiveTheme={this.toggleActiveTheme}
          />
          {children}
        </EmotionThemeProvider>
      </div>
    );
  }
}

export default ThemeProvider;
