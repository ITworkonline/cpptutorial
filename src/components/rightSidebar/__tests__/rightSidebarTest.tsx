import React from "react";
import renderer from "react-test-renderer";
import RightSidebar from "../index";
import { ThemeProvider } from "emotion-theming";
import { lightTheme } from "../../theme";

const mockToc = {
  items: [
    {
      title: "Level 1 heading 1",
      url: "level-1-heading-1",
      items: [
        {
          title: "Level 2 heading 1",
          url: "level-2-heading-1",
        },
      ],
    },
    {
      title: "Level 1 heading 1",
      url: "level-1-heading-1",
    },
  ],
};

describe("RightSidebar", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={lightTheme}>
          <RightSidebar toc={mockToc} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
