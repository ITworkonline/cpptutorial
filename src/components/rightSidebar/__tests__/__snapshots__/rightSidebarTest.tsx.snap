// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightSidebar renders correctly 1`] = `
.emotion-8 {
  width: 100%;
  border-right: 1px solid #ede7f3;
  height: 100vh;
  overflow: auto;
  position: fixed;
  padding-left: 24px;
  position: -webkit-sticky;
  position: -moz-sticky;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: #ffffff;
}

@media only screen and (max-width:50rem) {
  .emotion-8 {
    width: 100%;
    position: relative;
  }
}

.emotion-6 {
  margin-top: 32px;
}

.emotion-6 li {
  list-style-type: none;
  border-left: 1px solid #e6ecf1;
  border-left-color: rgb(230,236,241);
}

.emotion-6 li a {
  font-size: 12px;
  font-weight: 500;
  line-height: 1.5;
  padding: 7px 24px 7px 16px;
  color: #3B454E;
}

.emotion-0 {
  font-size: 10px;
  line-height: 1;
  font-weight: 700;
  text-transform: uppercase;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  padding: 7px 24px 7px 16px;
  border-left: 1px solid #e6ecf1;
  border-left-color: rgb(230,236,241);
  color: #3B454E;
}

.emotion-2 {
  list-style: none;
}

.emotion-2 a {
  color: #5c6975;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
  padding: 0.45rem 0 0.45rem 3rem;
  display: block;
  position: relative;
}

.emotion-2 a:hover {
  color: #3884C7 !important;
}

.emotion-2 a svg {
  float: right;
  margin-right: 1rem;
}

<aside
  className="emotion-8 emotion-9"
>
  <ul
    className="emotion-6 emotion-7"
  >
    <li
      className="emotion-0 emotion-1"
    >
      CONTENTS
    </li>
    <li
      className="emotion-2 emotion-3"
    >
      <a
        href="level-1-heading-1"
        to="level-1-heading-1"
      >
        Level 1 heading 1
      </a>
    </li>
    <li
      className="emotion-2 emotion-3"
    >
      <a
        href="level-1-heading-1"
        to="level-1-heading-1"
      >
        Level 1 heading 1
      </a>
    </li>
  </ul>
</aside>
`;
